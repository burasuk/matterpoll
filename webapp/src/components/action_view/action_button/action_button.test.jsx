import React from 'react';
import {shallow} from 'enzyme';

import ActionButton from 'components/action_view/action_button/action_button';

describe('components/action_button/ActionButton', () => {
    const baseProps = {
        currentUserId: 'user_id1',
        postId: 'post_id1',
        action: {
            id: 'action_id1',
            name: 'action_name',
        },
        voters: [
            'user_id1',
            'user_id2',
        ],
        actions: {doPostAction: jest.fn()},
    };
    test('should match snapshot', () => {
        const wrapper = shallow(<ActionButton {...baseProps}/>);
        expect(wrapper).toMatchSnapshot();
    });
    test('should match snapshot without voted', () => {
        const newProps = {
            ...baseProps,
            currentUserId: 'user_id3',
        };
        const wrapper = shallow(<ActionButton {...newProps}/>);
        expect(wrapper).toMatchSnapshot();
    });
});