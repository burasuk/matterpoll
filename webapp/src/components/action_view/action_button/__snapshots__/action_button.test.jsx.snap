// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/action_button/ActionButton should match snapshot 1`] = `
<button
  data-action-id="action_id1"
  key="action_id1"
  onClick={[Function]}
  style={
    Object {
      "background-color": "red",
    }
  }
>
  action_name
</button>
`;

exports[`components/action_button/ActionButton should match snapshot without voted 1`] = `
<button
  data-action-id="action_id1"
  key="action_id1"
  onClick={[Function]}
  style={Object {}}
>
  action_name
</button>
`;
